src/test/java/za/ac/cput/factory/BranchFactoryTest.java

```diff
@@ -0,0 +1,26 @@
+/*      BranchFactoryTest.java
+        Test for the Branch Factory
+        Author: Simphiwe Kahlana (220162891)
+        Date: 22 March 2023
+        */
+package za.ac.cput.factory;
+
+import org.junit.jupiter.api.Test;
+import za.ac.cput.domain.Branch;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class BranchFactoryTest {
+
+    @Test
+
+    public void testBranchFactory(){
+        Branch branch = BranchFactory.createBranch("Sim2891", "Paarl Auto",2008);
```

Comment:
Here, because the id is being generated in the factory and you removed the BranchId parameter, you would just have the 2 arguments passed in here (Branch name, year)

```suggestion
        Branch branch = BranchFactory.createBranch("Paarl Auto",2008);
```